AK/InsertionSort.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2022, Marc Luqué <marc.luque@outlook.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Concepts.h>
+#include <AK/StdLibExtras.h>
+
+namespace AK {
+
+// Standard Insertion Sort, with `end` inclusive!
+template<typename Collection, typename Comparator, typename T = decltype(declval<Collection>()[declval<int>()])>
+void insertion_sort(Collection& col, ssize_t start, ssize_t end, Comparator comparator) requires(Indexable<Collection, T>)
```

Comment:
So, we have a `size_t` to `ssize_t` implicit conversion based on the use below?

Reply:
I think you are misreading; all types used here are ssize.